javav2/example_code/redshift/src/main/java/com/example/redshift/scenario/RedshiftActions.java

```diff
@@ -0,0 +1,506 @@
+                for (Cluster cluster : clusterList) {
+                    if ("available".equals(cluster.clusterStatus())) {
+                        clusterReady = true;
+                        break;
+                    }
+                }
+
+                if (clusterReady) {
+                    logger.info(String.format("Cluster is available!"));
+                    return CompletableFuture.completedFuture(null);
+                } else {
+                    long elapsedTimeMillis = System.currentTimeMillis() - startTime;
+                    long elapsedSeconds = elapsedTimeMillis / 1000;
+                    long minutes = elapsedSeconds / 60;
+                    long seconds = elapsedSeconds % 60;
+                    System.out.printf("\rElapsed Time: %02d:%02d - Waiting for cluster...", minutes, seconds);
+                    System.out.flush();  // Ensure output is immediately flushed
+
+                    // Wait 1 second before the next status check
+                    return CompletableFuture.runAsync(() -> {
+                        try {
+                            TimeUnit.SECONDS.sleep(1);
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException("Error during sleep: " + e.getMessage(), e);
+                        }
+                    }).thenCompose(ignored -> checkClusterStatusAsync(clustersRequest, startTime));
+                }
+            }).exceptionally(exception -> {
+                throw new RuntimeException("Failed to get cluster status: " + exception.getMessage(), exception);
+            });
+    }
+    // snippet-end:[redshift.java2.describe_cluster.main]
+
+    /**
+     * Lists all databases asynchronously for the specified cluster, database user, and database.
+     * @param clusterId the identifier of the cluster to list databases for
+     * @param dbUser the database user to use for the list databases request
+     * @param database the database to list databases for
+     * @return a {@link CompletableFuture} that completes when the database listing is complete, or throws a {@link RuntimeException} if there was an error
+     */
+    public CompletableFuture<Void> listAllDatabasesAsync(String clusterId, String dbUser, String database) {
+        ListDatabasesRequest databasesRequest = ListDatabasesRequest.builder()
+            .clusterIdentifier(clusterId)
+            .dbUser(dbUser)
+            .database(database)
+            .build();
+
+        return getAsyncDataClient().listDatabases(databasesRequest)
+            .thenAccept(listDatabasesResponse -> {
+                listDatabasesResponse.databases().forEach(db -> {
+                    logger.info("The database name is {} ",db);
+                });
+            })
+            .exceptionally(exception -> {
+                throw new RuntimeException("Failed to list databases: " + exception.getMessage(), exception);
+            });
+    }
```

Comment:
Is there pagination for this and other list actions?

Reply:
Nice call - i will fix that